feat(ui): add DAG run state filter to grid view options

dheerajturaga opened the pull request:
Add filtering by DAG run state (queued, running, success, failed) to the grid view options panel, following the same pattern as the existing DagRuns.tsx implementation.

<img width="1747" height="882" alt="image" src="https://github.com/user-attachments/assets/a4c80447-668a-4a19-a120-02ad6670ed7b" />

<img width="1457" height="840" alt="image" src="https://github.com/user-attachments/assets/aa3039f4-6f5e-426e-ad76-2f3864cfedb6" />

dheerajturaga committed:
feat(ui): add DAG run state filter to grid view options

  Add filtering by DAG run state (queued, running, success, failed) to the grid view options panel, following the same pattern as the
  existing DagRuns.tsx implementation.

dheerajturaga committed:
Add testsgit add .

jscheffl commented on airflow-core/src/airflow/api_fastapi/core_api/datamodels/ui/common.py:
```diff
@@ -77,7 +77,7 @@ class GridRunsResponse(BaseModel):
     end_date: datetime | None
     run_after: datetime
-    state: TaskInstanceState | None
+    state: DagRunState | None
```
Uuups, good catch!

Important fix but... @bbovenzi /  @pierrejeambrun  this change is a bit of a breaking change. Must be prioritize this change for 3.1 release?

jscheffl replied:
@kaxil FYI might be we need to cherry-pick this line as "interface bug fix) (without the other feature but no harm to add it late) to 3.1.0 as the wrong type is on the public API

jscheffl reviewed (approved):
Looks good for me, tests are positive.

guan404ming reviewed (commented):
Overall looks good to me. Could you help take a look if this change is worked well with gantt chart? Thanks!

jason810496 reviewed (approved):
Thanks for the PR! LGTM on API side.

dheerajturaga committed:
Gantt chart filtering issue by applying the same pattern as PR #55845

The issue was that the Gantt chart wasn't using the same filters as the Grid view. When you set a DAG run state filter in the Options
  menu, it would filter the Grid view but not the Gantt chart. Now both views will be synchronized with the same filtering criteria.
